ssv/spectest/tests/runner/pre_consensus_justifications/duplicate_justification_signer.go

```diff
@@ -0,0 +1,115 @@
+package pre_consensus_justifications
+
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/ssv/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
+// DuplicateJustificationSigner tests a duplicate pre-consensus justification signer (error)
+func DuplicateJustificationSigner() tests.SpecTest {
+	ks := testingutils.Testing4SharesSet()
+
+	// duplicateSignerAndPrepareMsg will duplicate a justification signer
+	duplicateSignerAndPrepareMsg := func(obj *types.ConsensusData, id []byte) *qbft.SignedMessage {
+		if len(obj.PreConsensusJustifications) > 0 {
+			obj.PreConsensusJustifications[1] = obj.PreConsensusJustifications[0]
+		}
+
+		fullData, _ := obj.Encode()
+		root, _ := qbft.HashDataRoot(fullData)
+		msg := &qbft.Message{
+			MsgType:    qbft.ProposalMsgType,
+			Height:     1,
+			Round:      qbft.FirstRound,
+			Identifier: id,
+			Root:       root,
+		}
+		signed := testingutils.SignQBFTMsg(ks.Shares[1], 1, msg)
+		signed.FullData = fullData
+
+		return signed
+	}
+
+	expectedErr := "failed processing consensus message: invalid pre-consensus justification: duplicate signer"
+
+	return &tests.MultiMsgProcessingSpecTest{
+		Name: "pre consensus duplicate signer",
+		Tests: []*tests.MsgProcessingSpecTest{
+			{
+				Name:   "sync committee aggregator selection proof",
+				Runner: decideFirstHeight(testingutils.SyncCommitteeContributionRunner(ks)),
```

Comment:
why do we `decideFirstHeight`?
Not sure this is bad, but seems not necessary here I think

Reply:
doesn't make a difference, could be on height 0 as well